builder/bootloader.py

```diff
@@ -54,6 +55,41 @@ def get_suitable_optiboot_binary(framework_dir, board_config):
     return bootloader_path.replace(".hex", "_BIGBOOT.hex")
 
 
+def get_suitable_urboot_binary(framework_dir, board_config):
+    mcu = board_config.get("build.mcu", "").lower()
+    f_cpu = re.sub("[^0-9]", "", board_config.get("build.f_cpu", "16000000L"))
```

Comment:
Wouldn't it be easier to just drop the closing `L` by using list comprehension:
```python
board_config.get("build.f_cpu", "16000000L")[:-1]
```
Also, if we cast the `f_cpu` variable to integer in this line, we won't need a lot of `int(f_cpu)` calls below.

Reply:
I'm not very good with python, so I don't know all the lever tricks. There may or may not be a capital `L` or `UL` at the of the number. If there is a more efficient way to remove trailing letters, I'm all ears!

Reply:
IIRC, there are no `UL` literals in PlatformIO dev-platforms, but anyway you can use `board_config.get("build.f_cpu", "16000000L").strip("UL")` instead of importing the regex module.

Reply:
I've just pushed a fix for this as well. Please review 🙂